base/array.jl

```diff
@@ -1251,10 +1251,11 @@ julia> findnext(A,3)
 function findnext(A, start::Integer)
     for i = start:length(A)
         if A[i] != 0
-            return i
```

Comment:
I do like the `for else` feature but isn't `v = for ...; break i; else j end` always equivalent to `for ...; v = i; break; else v = j end` (with the correct scope of `v`)?
The latter seems much easier to understand than the former. It is similar to the return value of `if else` but in that case at least the return value is easier to fine.